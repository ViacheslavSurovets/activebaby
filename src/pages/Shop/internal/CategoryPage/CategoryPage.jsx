import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import { selectCollection, selectCollections } from '@redux/shop/shop.selecors';
import { CollectionOverviewItem, CollectionPreviewItem, Portal } from '@components';
import {
  CategoryPageContainer,
  CategorySectionTitle,
  CategorySectionContainer,
  CategoryItemTitle,
  CategoryItemWrapper,
} from './styles';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { selectShowPortalPreviewItem } from '@redux/previewItem/previewItem.selectors';



const CategoryPage = ( { collection, showPortalPreviewItem } ) => {
  const { t, i18n } = useTranslation ();
  console.log ( collection );
  return (
    <CategoryPageContainer>
      <CategorySectionTitle>{ t ( collection.title ) }</CategorySectionTitle>

      {
        showPortalPreviewItem && <Portal component={CollectionPreviewItem} />
      }

      {
        collection.items.map ( ( section, idx ) =>
          <CategorySectionContainer key={ idx }>
            <CategoryItemTitle>{ t ( section.title ).toUpperCase () }</CategoryItemTitle>
            <CategoryItemWrapper>
              {
                section.types.map ( (item, index) =>
                  <Fragment key={index}>
                    <CollectionOverviewItem item={ item } info={section.info}/>
                  </Fragment>
                )
              }
            </CategoryItemWrapper>
          </CategorySectionContainer>
        )
      }
    </CategoryPageContainer>
  );
};

const mapStateToProps = ( state, ownProps ) => ({
  collection: selectCollection ( ownProps.match.params.collectionId ) ( state ),
  showPortalPreviewItem: selectShowPortalPreviewItem(state),
  collections: selectCollections ( state )
});




export default connect ( mapStateToProps ) ( CategoryPage );


CategoryPage.propTypes = {
  collection: PropTypes.object,
  showPortalPreviewItem: PropTypes.bool
};

