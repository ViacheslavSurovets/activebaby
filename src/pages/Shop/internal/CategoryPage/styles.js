import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { commonStyleTextThreeRem, commonStyleTextTwoRem, commonLinkStyle, commonStyleTextFourRem } from '@commonStyles';
import { tablet, mobile } from '@utils/media';

export const CategoryPageContainer = styled.section`
  width: 95%;
  margin:0 auto;
  display: flex;
  flex-direction: column;
`;

export const CategorySectionTitle = styled.span`
  ${ commonStyleTextFourRem };
`;


export const CategorySectionContainer = styled.section`
  display:flex;
  flex-direction: column;
  width: 100%;
  margin-top: 5rem;
`;

export const CategoryItemTitle = styled.span`
  ${ commonStyleTextThreeRem };
`;

export const CategoryItem = styled.div`
  display: flex;
  flex-direction: column;
  flex: 0 1 250px;
  height: 30vh;
  margin: 1rem;
  transition: all 1s;

  background-color: ghostwhite;
  &:hover{
    transform: scale(1.05);
    transition: all 1s;
    box-shadow: 4px 4px 4px -3px ${ ( { theme } ) => theme.black };
  }
`;


export const CategoryItemImageLink = styled ( Link )`
  ${ commonLinkStyle };
  background-image: url(${ props => props.background });
  background-size: 100% 100%;
  flex: 1;
  margin: .5rem;
`;

export const CategoryItemWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  ${tablet},
  ${mobile}{
    justify-content: center;
  }
`;

export const CategoryItemText = styled.span`
  ${ commonStyleTextTwoRem };
  font-weight: bold;
`;

export const CategoryItemPriceText = styled.span`
  font-size: 1.6rem;
  font-weight: lighter;
`;

export const CategoryItemInfoWrapper = styled.div`
  display: flex;
  align-items: center; 
  height: 3rem;
  justify-content: space-between;
`;

export const CategoryItemCustomButtonWrapper = styled.div`
  &:hover {
   svg {
        fill: ${ ( { theme } ) => theme.header.cartColor };
      };
  }
`;
