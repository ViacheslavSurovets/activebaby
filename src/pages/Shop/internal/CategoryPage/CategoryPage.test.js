import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import CategoryPage from './CategoryPage';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('CategoryPage', ()=> {
  it('CategoryPage Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <CategoryPage  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});
