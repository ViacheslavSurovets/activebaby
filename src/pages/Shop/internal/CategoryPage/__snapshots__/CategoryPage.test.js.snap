// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryPage CategoryPage Snapshot 1`] = `ShallowWrapper {}`;
